### PR TITLE
[auth] make id-token optional

### PR DESCRIPTION
**Pull request by chaudyg**
This make id-token optional.

2 scenarios:
- Background services (like the dualwriter syncer) will run with the underlying grafana permissions. In the request from grafana to unistore, there will be only the underlying access token from grafana
- A service calling grafana. In this use case, apiserver caller will only have an access token an no id token. In this scenario, the permission is enforced by grafana, and the request from grafana to unistore is also done using grafana access_token with grafana permissions.

### CODE REVIEW

File: pkg/storage/unified/resource/client.go [hunk 1]
```diff
@@ -128,29 +174,17 @@ func NewCloudResourceClient(tracer tracing.Tracer, conn *grpc.ClientConn, cfg au
...
-		token, _, err := createInternalToken(staticRequester)
-		if err != nil {
-			return "", fmt.Errorf("failed to create internal token: %w", err)
-		}
-
-		staticRequester.IDToken = token
-		return token, nil
-	}
-
 	return "", fmt.Errorf("id-token not found")
```
Comment (chaudyg) [inline pkg/storage/unified/resource/client.go:188]: This should be nil

**Review by ryantxu — APPROVED** [review]
LGTM -- i wonder if our UI in general should support this, but not sure

**Review by gamab — APPROVED** [review]

File: pkg/storage/unified/resource/client.go [hunk 2]
```diff
@@ -104,10 +104,35 @@ func NewGRPCResourceClient(tracer tracing.Tracer, conn *grpc.ClientConn) (Resour
...
+	// This is a workaround for StaticRequester not having a signed ID token.
+	if staticRequester, ok := authInfo.(*identity.StaticRequester); ok {
+		token, _, err := createInternalToken(staticRequester)
+		if err != nil {
+			return "", fmt.Errorf("failed to create internal token: %w", err)
+		}
+
+		staticRequester.IDToken = token
+		return token, nil
+	}
```
[the remark this answers is not in the document]
Comment (chaudyg) [reply pkg/storage/unified/resource/client.go:127]: I think it's needed for tests

[the remark this answers is not in the document]
Comment (gamab) [reply pkg/storage/unified/resource/client.go:127]: Can we modify the tests to use claims and have only one extractor?
I don't know how much work this means. So ignore if you want :)

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**